source/gridfs/gridfs-spec.md

```diff
@@ -649,6 +649,32 @@ orphaned chunks with files_id equal to id before raising the error.
 
 If a networking or server error occurs, drivers MUST raise an error.
 
+### File deletion by filename
+
+To rename all revisions of a stored file with the specified filename, drivers SHOULD provide the method
+`delete_by_name`:
+
+```javascript
+class GridFSBucket {
+
+  /**
+   * Deletes all stored files with the specified @filename from a GridFS bucket.
+   */
+  void delete_by_name(string filename);
+
+}
+```
+
+This method is an optimisation over deleting each revision of a stored file individually.
+
+**Implementation details:**
+
+Drivers MUST first find the `_id` field of all files collection documents with the given filename. Drivers MUST then
+delete all chunks with `files_id` in the found ids that was just deleted. Drivers MUST then delete all files collection
```

Comment:
Wouldn't we want to delete the files documents first and then the chunks? Doing the opposite increases the chance of reading a phantom corrupt file. This is the same reason we always upload the chunks before saving the files document.

Reply:
You are correct. Files should be deleted before chunks to ensure there is no partial files. But if the operation that deletes chunks fails, then there is orphan chunks.

Spec updated.